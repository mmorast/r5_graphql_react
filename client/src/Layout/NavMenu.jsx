import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'


const links = [
  {to: "/", text: "Home"},
  {to: "/posts", text: "Posts"},
]

const NavMenu = props => {

  return (
    <NavDiv>
      {links.map((link, idx) => {
        return <StyledLink key={`link-${idx}`} to={link.to}>{link.text}</StyledLink>
      })}
    </NavDiv>
  )
}

export default NavMenu

const NavDiv = styled.nav`
  padding: 10px;
`

const StyledLink = styled(Link)`
  padding: 5px;
  color: white;
`
