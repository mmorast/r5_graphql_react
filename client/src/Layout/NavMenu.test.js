import React from 'react'
import {shallow} from 'enzyme'
import NavMenu from './NavMenu'


test("Renders the NavMenu", () => {
  let actual = shallow(<NavMenu/>)
  expect(actual).toMatchSnapshot()
})

