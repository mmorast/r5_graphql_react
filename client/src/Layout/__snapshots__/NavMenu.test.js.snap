// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the NavMenu 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Uncontrolled(Navbar)
    bsStyle="default"
    collapseOnSelect={true}
>
    <NavbarHeader>
        <NavbarBrand>
            <Link
                replace={false}
                to="/"
            >
                R5+GraphQL+React
            </Link>
        </NavbarBrand>
        <NavbarToggle />
    </NavbarHeader>
    <NavbarCollapse>
        <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={false}
            stacked={false}
        >
            <LinkContainer
                activeClassName="active"
                exact={false}
                replace={false}
                strict={false}
                to="/posts"
            >
                <NavItem
                    active={false}
                    disabled={false}
                >
                    Posts
                </NavItem>
            </LinkContainer>
        </Nav>
        <Nav
            bsClass="nav"
            justified={false}
            pullLeft={false}
            pullRight={true}
            stacked={false}
        >
            <LinkContainer
                activeClassName="active"
                exact={false}
                replace={false}
                strict={false}
                to="/login"
            >
                <NavItem
                    active={false}
                    disabled={false}
                >
                    Log In
                </NavItem>
            </LinkContainer>
            <LinkContainer
                activeClassName="active"
                exact={false}
                replace={false}
                strict={false}
                to="/signup"
            >
                <NavItem
                    active={false}
                    disabled={false}
                >
                    Sign Up
                </NavItem>
            </LinkContainer>
        </Nav>
    </NavbarCollapse>
</Uncontrolled(Navbar)>,
  "nodes": Array [
    <Uncontrolled(Navbar)
      bsStyle="default"
      collapseOnSelect={true}
>
      <NavbarHeader>
            <NavbarBrand>
                  <Link
                        replace={false}
                        to="/"
                  >
                        R5+GraphQL+React
                  </Link>
            </NavbarBrand>
            <NavbarToggle />
      </NavbarHeader>
      <NavbarCollapse>
            <Nav
                  bsClass="nav"
                  justified={false}
                  pullLeft={false}
                  pullRight={false}
                  stacked={false}
            >
                  <LinkContainer
                        activeClassName="active"
                        exact={false}
                        replace={false}
                        strict={false}
                        to="/posts"
                  >
                        <NavItem
                              active={false}
                              disabled={false}
                        >
                              Posts
                        </NavItem>
                  </LinkContainer>
            </Nav>
            <Nav
                  bsClass="nav"
                  justified={false}
                  pullLeft={false}
                  pullRight={true}
                  stacked={false}
            >
                  <LinkContainer
                        activeClassName="active"
                        exact={false}
                        replace={false}
                        strict={false}
                        to="/login"
                  >
                        <NavItem
                              active={false}
                              disabled={false}
                        >
                              Log In
                        </NavItem>
                  </LinkContainer>
                  <LinkContainer
                        activeClassName="active"
                        exact={false}
                        replace={false}
                        strict={false}
                        to="/signup"
                  >
                        <NavItem
                              active={false}
                              disabled={false}
                        >
                              Sign Up
                        </NavItem>
                  </LinkContainer>
            </Nav>
      </NavbarCollapse>
</Uncontrolled(Navbar)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <NavMenu />,
    "_instance": null,
    "_newState": null,
    "_rendered": <Uncontrolled(Navbar)
      bsStyle="default"
      collapseOnSelect={true}
>
      <NavbarHeader>
            <NavbarBrand>
                  <Link
                        replace={false}
                        to="/"
                  >
                        R5+GraphQL+React
                  </Link>
            </NavbarBrand>
            <NavbarToggle />
      </NavbarHeader>
      <NavbarCollapse>
            <Nav
                  bsClass="nav"
                  justified={false}
                  pullLeft={false}
                  pullRight={false}
                  stacked={false}
            >
                  <LinkContainer
                        activeClassName="active"
                        exact={false}
                        replace={false}
                        strict={false}
                        to="/posts"
                  >
                        <NavItem
                              active={false}
                              disabled={false}
                        >
                              Posts
                        </NavItem>
                  </LinkContainer>
            </Nav>
            <Nav
                  bsClass="nav"
                  justified={false}
                  pullLeft={false}
                  pullRight={true}
                  stacked={false}
            >
                  <LinkContainer
                        activeClassName="active"
                        exact={false}
                        replace={false}
                        strict={false}
                        to="/login"
                  >
                        <NavItem
                              active={false}
                              disabled={false}
                        >
                              Log In
                        </NavItem>
                  </LinkContainer>
                  <LinkContainer
                        activeClassName="active"
                        exact={false}
                        replace={false}
                        strict={false}
                        to="/signup"
                  >
                        <NavItem
                              active={false}
                              disabled={false}
                        >
                              Sign Up
                        </NavItem>
                  </LinkContainer>
            </Nav>
      </NavbarCollapse>
</Uncontrolled(Navbar)>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NavMenu />,
}
`;
